### PR TITLE
windows: Support alt install paths, fit and finish

### LINKED ISSUE #2776: Support custom install dir in Windows installer
Hi,
Is there an option to install the windows app NOT on C:\ drive ?

e.g.: E:\Program Files\.....

### LINKED ISSUE #6609: windows: auto updater takes a long time to restart
### What is the issue?

After clicking on "Restart to update" in the Windows app tray icon, it takes 10-15 seconds for the app to restart.

### OS

Windows

### GPU

_No response_

### CPU

_No response_

### Ollama version

_No response_

### LINKED ISSUE #6725: Incorrect AppDir when creating banner script (Preview)
Upon installing Ollama (Preview, 0.3.10.0) on Windows*, I noticed an error in the log file:

```
time=2024-09-10T05:45:41.543-05:00 level=INFO source=logging.go:50 msg="ollama app started"
time=2024-09-10T05:45:41.564-05:00 level=INFO source=store.go:96 msg="wrote store: C:\\Users\\desco\\AppData\\Local\\Ollama\\config.json"
time=2024-09-10T05:45:41.581-05:00 level=INFO source=store.go:96 msg="wrote store: C:\\Users\\desco\\AppData\\Local\\Ollama\\config.json"
time=2024-09-10T05:45:41.582-05:00 level=INFO source=server.go:176 msg="unable to connect to server"
time=2024-09-10T05:45:41.583-05:00 level=INFO source=server.go:135 msg="starting server..."
time=2024-09-10T05:45:41.963-05:00 level=INFO source=server.go:121 msg="started ollama server with pid 7604"
time=2024-09-10T05:45:41.963-05:00 level=INFO source=server.go:123 msg="ollama server logs C:\\Users\\desco\\AppData\\Local\\Ollama\\server.log"
time=2024-09-10T05:45:43.695-05:00 level=WARN source=lifecycle.go:51 msg="Failed to launch getting started shell: getting started banner script error CreateFile C:\\Users\\desco\\AppData\\Local\\Programs\\Ollama\\ollama_welcome.ps1: The system cannot find the path specified."
```

* Note: Installer launched via `ollamasetup.exe /DIR="H:/ollama"`

The app.log file is located in `%LOCALAPPDATA%\Ollama`, but the error message indicates the AppDir is erroneously set `%LOCALAPPDATA%\Programs\Ollama`. It's worth noting that `ollama_welcome.ps1` **was** successfully installed to `H:\ollama\ollama_welcome.ps1`, which correctly follows the user-specified install directory.

Unfortunately, I have little to no proficiency in Go, but I figured I should point it out regardless.

https://github.com/ollama/ollama/blob/83a9b5271a68c7d1f8443f91c8d8b7d24ab581a9/app/lifecycle/getstarted_windows.go#L15

<hr>

### Environment Info

- **OS**: Windows 11 Pro, build 22631
- **CPU**: Ryzen 5 3600
- **GPU**: RTX 4060

- **Installer version**: Preview/0.3.10.0
SHA256: 3BE19A085685324066762F33C46C4A1121F27E7A1EA9B441D0BECF57DBB34375

### PR DESCRIPTION
**Pull request by dhiltgen**
- Document alternate install location instructions for binaries and model, and fix path resolution relative to executable
- Pop up progress UI for upgrades (automatic, with cancel button).
- Expose non-default port in menu to disambiguate multiple instances.

Fixes #6725 
Fixes #2776 
Fixes #6609

### CODE REVIEW

**Review by vignesh1507 — COMMENTED** [review]
Looks good to me and thx for updating the code, appreciate the effort @dhiltgen

**Comment by CarlKenner** [comment]
Are you going to fix the installer to install to a sane location or ask the user where to install it? You shouldn't have to be an "advanced user" to do what literally every other software installer does for you.
Forcing the installation of huge AI models and huge CUDA libraries to the smallest drive is a bad idea.

File: app/tray/wintray/menus.go [hunk 1]
```diff
@@ -8,18 +8,28 @@ import (
...
+	updateMenuID
+	separatorMenuID
+	diagLogsMenuID
+	diagSeparatorMenuID
+	quitMenuID
 )
 
 func (t *winTray) initMenus() error {
+	port := envconfig.Host().Port()
+	if port != "11434" {
```
Comment (jmorganca) [inline app/tray/wintray/menus.go:28]: I'd leave this to a future settings page just to avoid too many potential items in the list

**Review by jmorganca — COMMENTED** [review]
Overall looks good. I'd omit the change for the menubar for now though

File: app/lifecycle/server.go [hunk 2]
```diff
@@ -18,11 +18,17 @@ func getCLIFullPath(command string) string {
 	var cmdPath string
 	appExe, err := os.Executable()
 	if err == nil {
+		// Check both the same location as the tray app, as well as ./bin
 		cmdPath = filepath.Join(filepath.Dir(appExe), command)
 		_, err := os.Stat(cmdPath)
 		if err == nil {
 			return cmdPath
 		}
+		cmdPath = filepath.Join(filepath.Dir(appExe), "bin", command)
```
Comment (jmorganca) [inline app/lifecycle/server.go:27]: Would the `ollama` cli be in either directory?

**Review by jmorganca — APPROVED** [review]

**Review by jmorganca — APPROVED** [review]

File: app/lifecycle/server.go (hunk 2 above)
Comment (dhiltgen) [reply app/lifecycle/server.go:27]: Just a little safeguarding for potentially converging to a common layout between windows and the other OSes where we put the CLI in ./bin relative to the "install dir". It searches both locations so it's safe

**Comment by TangentFoxy** [comment]
Is there an estimate for when the release that includes this patch will be?

**Pull request merged**